mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java

```diff
@@ -59,918 +55,1054 @@
+            updateTextViewColor(mLabelView, widget.getLabelColor());
+            updateIcon(mIconView, widget);
+            if (widget.hasItem()) {
+                int progress;
+                if (widget.getItem().getType().equals("Color") || "Color".equals(widget.getItem().getGroupType())) {
+                    try {
+                        progress = widget.getItem().getStateAsBrightness();
+                    } catch (IllegalStateException e) {
+                        progress = 0;
+                    }
                 } else {
-                    return TYPE_VIDEO;
+                    progress = widget.getItem().getStateAsFloat().intValue();
                 }
+                mSeekBar.setProgress(progress);
             } else {
-                return TYPE_VIDEO;
+                mSeekBar.setProgress(0);
             }
-        } else if (openHABWidget.getType().equals("Webview")) {
-            return TYPE_WEB;
-        } else if (openHABWidget.getType().equals("Colorpicker")) {
-            return TYPE_COLOR;
-        } else {
-            return TYPE_GENERICITEM;
+            mBoundItem = widget.getItem();
+        }
+
+        @Override
+        public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+        }
+
+        @Override
+        public void onStartTrackingTouch(SeekBar seekBar) {
+            Log.d(TAG, "onStartTrackingTouch position = " + seekBar.getProgress());
+        }
+
+        @Override
+        public void onStopTrackingTouch(SeekBar seekBar) {
+            Log.d(TAG, "onStopTrackingTouch position = " + seekBar.getProgress());
+            Util.sendItemCommand(mHttpClient, mBoundItem, String.valueOf(seekBar.getProgress()));
         }
     }
 
-    public void setOpenHABBaseUrl(String baseUrl) {
-        openHABBaseUrl = baseUrl;
+    public static class ImageViewHolder extends ViewHolder {
+        private final MySmartImageView mImageView;
+        private final Point mScreenSize = new Point();
+        private int mRefreshRate;
+
+        ImageViewHolder(LayoutInflater inflater, ViewGroup parent,
+                MyAsyncHttpClient httpClient, ConnectionInfo connection) {
+            super(inflater, parent, R.layout.openhabwidgetlist_imageitem, httpClient, connection);
+            mImageView = itemView.findViewById(R.id.imageimage);
+
+            WindowManager wm = (WindowManager) parent.getContext().getSystemService(Context.WINDOW_SERVICE);
+            wm.getDefaultDisplay().getSize(mScreenSize);
+
+        }
+
+        @Override
+        public void bind(OpenHABWidget widget) {
+            View parent = (View) itemView.getParent();
+            // We scale the image at max 90% of the available height
+            int parentWidth = parent != null ? parent.getWidth() : 0;
+            int parentHeight = parent != null ? parent.getHeight() : 0;
+            mImageView.setMaxSize(parentWidth > 0 ? pare
```

Comment:
Isn't the item type always Image here?

Reply:
Yes, it is. It was taken straight from the old code, but as it's redundant, I'll remove it.

Reply:
Removed; also replaced the startsWith() with a more meaningful check.